nvflare/private/fed/client/client_json_config.py

```diff
@@ -49,20 +50,35 @@ def __init__(self, config_file_name: str, args, app_root: str, kv_list=None, exc
         base_pkgs = FL_PACKAGES
         module_names = FL_MODULES
 
+        if kv_list:
+            assert isinstance(kv_list, list), "cmd_vars must be list, but got {}".format(type(kv_list))
+            self.cmd_vars = parse_vars(kv_list)
+        else:
+            self.cmd_vars = {}
+
+        # determine the values of variables that can be used in job config.
+        sp_scheme = args.sp_scheme
+        sp_target = args.sp_target
+        sp_url = f"{sp_scheme}://{sp_target}"
```

Comment:
should SP thing come from Overseer? @IsaacYangSLA

Reply:
I thought about that. But this is the CJ that was just started by CP, which has the CP from Overseer. Though it is possible, in theory, that when CJ is started, the SP is just changed, but I don't think we need to worry about it.

Besides it's impossible to get SP from Overseer in this case because the job config parsing happens at the very beginning of the process before SP CB happens.